src/main/kotlin/org/rust/lang/core/resolve2/FacadeResolve.kt

```diff
@@ -136,23 +136,17 @@ private fun ModData.processMacros(
 ): Boolean {
     val isQualified = macroPath == null || macroPath is RsPath && macroPath.qualifier != null
 
-    for ((name, perNs) in visibleItems.entriesWithName(processor.name)) {
-        for (visItem in perNs.macros) {
-            val isLegacyMacroDeclaredInSameMod = !isQualified && legacyMacros[name].orEmpty().any {
-                (!isAttrOrDerive || it is ProcMacroDefInfo) && it.path.parent == path
-            }
-            if (isLegacyMacroDeclaredInSameMod) {
-                // Resolve order for unqualified macros:
-                // - textual macros in same mod
-                // - scoped macros (imported by `use`)
-                // - textual macros
-                continue
-            }
-            val macro = visItem.scopedMacroToPsi(defMap, project) ?: continue
-            val visibilityFilter = visItem.visibility.createFilter(project)
-            if (processor(name, macro, visibilityFilter)) return true
+    val stop = processScopedMacros(processor, defMap, project) { name ->
+        val isLegacyMacroDeclaredInSameMod = !isQualified && legacyMacros[name].orEmpty().any {
+            (!isAttrOrDerive || it is ProcMacroDefInfo) && it.path.parent == path
         }
+        // Resolve order for unqualified macros:
+        // - textual macros in same mod
+        // - scoped macros (imported by `use`)
+        // - textual macros
+        !isLegacyMacroDeclaredInSameMod
     }
+    if (stop) return true
```

Comment:
Is there a case when this line affects the name resolution result or it's only about performance? If yes, can we add the corresponding test?

Reply:
This is common convention in resolve code to stop processing if `processor` returns true for some `ScopeEntry`. Actually code here hasn't changed, I just extracted `processScopedMacros` function. It can affect name resolution and there are already some tests which check it, e.g. ```RsPackageLibraryResolveTest#`test import wins extern crate import` ```

Reply:
Or, sorry, I missed `return true` in previous version